### PR TITLE
Implement editGarnConfig to spin up a pre-configured vscodium

### LINKED ISSUE #194: garn edit
It's quite annoying and non-obvious to set up your text editor correctly in order to get LSP/IDE goodness for garn.ts files. 

We could help out somehow. For instance providing a `garn.edit` Project that has an executable that is vscodium with the appropriate settings, so users could run `garn run edit`

### PR DESCRIPTION
**Pull request by soenkehahn**
Fixes #194.

### CODE REVIEW

**Review by jkarni — CHANGES_REQUESTED** [review]
Cool! It isn't really working very well for me, though. I manage to start up the editor, but I get "Cannot find module 'http://localhost:8777/mod.ts' or its corresponding type declarations.ts(2307)" on startup (same with `garn.io` domain.

File: ts/edit.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      "editor.defaultFormatter": "denoland.vscode-deno",
+    },
+    "update.mode": "none",
+  };
+  return writeTextFile("vscodium-config", JSON.stringify(settings, null, 2));
+};
+
+/**
+ * A preconfigure state.vscdb sqlite file that will prevent vscodium to open a
+ * welcome tab for the deno extension.
```
Comment (jkarni) [inline ts/edit.ts:43]: I got the welcome tab.

**Comment by soenkehahn** [comment]
@jkarni: Hmm, for some reason this doesn't seem to work as expected on your machine. Both the deno caching and putting the settings in place. It works on all the machines I've tested on (two linux, one mac). Maybe we can debug this together?

File: website/src/docs/getting_started.mdx [hunk 2]
```diff
@@ -90,7 +90,24 @@ without having to install any language-specific tools or learn how to use them.
...
 `garn.ts` files are powered by [deno](https://deno.com/).
 A lot of the convenience and power for editing your `garn.ts` files comes
-from having a working Deno LSP. Installing `deno` is easy with `garn` itself:
+from having a working Deno LSP.
+
+### `editGarnConfig`
+
+`garn` offers an `Executable` `editGarnConfig` that you can add to your
+`garn.ts` file and that you can run to spin up a vscodium editor that is
+pre-configured for editing `garn.ts` files. This is the easiest way to get a
```
[the remark this answers is not in the document]
Comment (soenkehahn) [reply website/src/docs/getting_started.mdx]: I reworked this section in 55bec1c0. What do you think?

Regarding both macs: I tested this only on an arm mac. I propose to ignore intel macs for the time being (until someone complains). We don't have a binary cache for intel macs. So even if everything works *correctly*, everything will be terribly slow, including installing `garn` and building any project. (And testing this editor.) I think -- if we decide intel macs are important enough to support well -- the next step is to set up a builder on garnix.io for that. In the meantime we should add a warning to the website that this doesn't work well on intel macs.

File: website/src/docs/getting_started.mdx [hunk 3]
```diff
@@ -17,32 +17,37 @@ The easiest way to install <Garn/> is with the following command:
...
-  ```
+
+```
 extra-substituters = https://cache.garnix.io
 extra-trusted-public-keys = cache.garnix.io:CTFPyKSLcx5RMJKfLo5EEPUObbA78b0YQ2DTCJXqr9g=
-  ```
-  To `/etc/nix/nix.conf` (note that you'll need sudo).
+```
+
+To `/etc/nix/nix.conf` (note that you'll need sudo).
```
Comment (jkarni) [inline website/src/docs/getting_started.mdx:35]: Maybe doesn't need to be now, but just occurred to me that we need to tell users to restart the daemon after this change.

**Review by jkarni — APPROVED** [review]

File: website/src/docs/getting_started.mdx (hunk 3 above)
Comment (soenkehahn) [reply website/src/docs/getting_started.mdx:35]: Good point! I think we don't though. At least `nix show-config` shows the the cache without restarting. Although we should make sure that the cache is actually being used.

Comment (soenkehahn) [reply website/src/docs/getting_started.mdx:35]: Yeah, never mind. `nix show-config` is just lying.

**Pull request merged**